content/intro-to-onchain.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Intro to on-chain development
+objectives:
+- Understand how Solana on-chain programs work
+- Describe the structure and operation of on-chain program
+- Build a basic program
+---
+
+# Summary
+
+- **On chain programs** are programs that run on Solana. 
+- Programs have one or more **instruction handlers**. Instruction handlers are functions that process the **instructions** in Solana transactions.
+- Instruction handlers write their data to Solana **accounts** and then either succeed or fail. 
+- Solana programs are most commonly written in **Rust**, typically with the **Anchor** framework.
+- Anchor programs have IDLs, small JSON files which can be used to automatically create JS/TS code to build instructions for clients.  
+
+# Overview
+
+Each Solana cluster (mainnet-beta, testnet, devnet, localnet) is effectively a single computer with globally syncronized state. The programs that run on Solana - the ones that create tokens, swap tokens, art marketplaces, escrows, market makers, DePIN apps, auctions, retail payments platforms, etc - are called **Solana apps**.
+
+The most popular way to build onchain apps is using **Rust** language and the **Anchor** framework. Frameworks for other languages, including **TypeScript**, **Python** and **Move** are coming soon. 
+
+These frameworks implement common security checks automatically, and handle common tasks like:
+ - sending incoming instructions to the right instruction handlers
+ - deserialising data from incoming transactions
```

Comment:
Would prefer the American spelling for deserialize

Reply:
Might consider running this through something like Grammarly with those sorts of things in mind